pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -151,5 +140,13 @@ func (b *Balancer) AddService(name string, handler http.Handler, weight *int) {
 	if weight != nil {
 		w = *weight
 	}
-	b.handlers = append(b.handlers, &namedHandler{Handler: handler, name: name, weight: w})
+	if w <= 0 { // non-positive weight is meaningless
```

Comment:
I understand why you're doing that, but I am not sure we can afford it.
There was nothing preventing users adding a service w <=0 before, and I do not think it was ignored. In fact, if i understand correctly the old nextServer algorithm, even servers with a weight of 0 were taken into account when picking a server. We only had an error if absolutely all the weights were equal to 0 at first glance.
Usually the least we could do is document that AddService will ignore a service if its weight is <=0, but I don't think that would be enough. if we don't want to break our existing users, we need to make the whole algorithm valid within the same parameters as the old algorithm.

But maybe I am wrong, and maybe the old nextServer algorithm actually ignored any service with a weight <=0, even thought it was added to the list of services. We need to make sure of that. I will double-check and ask around.

Reply:
OK, nevermind that last remark. Even if the old algorithm was not clear on the matter, the intent was indeed to ignore any service with a weight <=0. So let's document AddService in consequence. I will add a suggestion in a minute.

Reply:
Added